### PR TITLE
[DROOLS-5617] Refactor MiningModel source generation

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @mariofusco @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-5617

With this PR
1) *nested model* sources are not put inside the MiningModel (container) sources map anymore
2) *nested models* are add to 
`KiePMMLMiningModelWithSources.nestedModels`
list as "KiePMMLModel+HasSources" models
3) Such list will be populated during code-generation and read during compilation

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactory.java [hunk 1]
```diff
@@ -85,14 +86,16 @@ public static KiePMMLSegmentation getSegmentation(final DataDictionary dataDicti
...
+                                                                final List<KiePMMLModel> nestedModels) {
         logger.debug("getSegmentationSourcesMap {}", segmentation);
         final String packageName = getSanitizedPackageName(parentPackageName + "." + segmentationName);
         final Map<String, String> toReturn = getSegmentsSourcesMap(packageName,
                                                                   dataDictionary,
                                                                   transformationDictionary,
                                                                   segmentation.getSegments(),
-                                                                  kBuilder);
+                                                                  kBuilder,
+                                                                   nestedModels);
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactory.java]: Fix formatting

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactoryTest.java [hunk 2]
```diff
@@ -77,13 +81,16 @@ public void getKiePMMLMiningModel() {
...
         final Map<String, String> retrieved = KiePMMLMiningModelFactory.getKiePMMLMiningModelSourcesMap(
                                                                                                        DATA_DICTIONARY,
                                                                                                        TRANSFORMATION_DICTIONARY,
                                                                                                        MINING_MODEL,
                                                                                                        packageName,
-                                                                                                       KNOWLEDGE_BUILDER);
+                                                                                                       KNOWLEDGE_BUILDER,
+                                                                                                       nestedModels);
         assertNotNull(retrieved);
+        assertFalse(nestedModels.isEmpty());
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactoryTest.java]: Can you please add an assertion related to number of expected nested models?

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactoryTest.java [hunk 3]
```diff
@@ -52,12 +56,15 @@ public void getSegmentation() {
...
         final Map<String, String> retrieved = KiePMMLSegmentationFactory.getSegmentationSourcesMap(packageName,
                                                                                                    DATA_DICTIONARY,
                                                                                                    TRANSFORMATION_DICTIONARY,
                                                                                                    MINING_MODEL.getSegmentation(),
                                                                                                    segmentationName,
-                                                                                                   KNOWLEDGE_BUILDER);
+                                                                                                   KNOWLEDGE_BUILDER,
+                                                                                                   nestedModels);
         assertNotNull(retrieved);
+        assertFalse(nestedModels.isEmpty());
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactoryTest.java]: Can you please add an assertion related to number of expected nested models?

**Review by danielezonca — APPROVED** [review]
LGTM

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactoryTest.java (hunk 3 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactoryTest.java]: @danielezonca 
Done

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactoryTest.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactoryTest.java]: @danielezonca 
Done

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactory.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLSegmentationFactory.java]: @danielezonca 
Done

**Review by jiripetrlik — APPROVED** [review]
Looks good to me.

**Comment by gitgabrio** [comment]
@danielezonca Hi! Could you please merge this, squashing commits? Failure in DB are unrelated. Thanks!

**Pull request merged**